metadata/images.go

```diff
@@ -254,12 +255,7 @@ func (s *imageStore) Delete(ctx context.Context, name string, opts ...images.Del
 			return errors.Wrapf(errdefs.ErrNotFound, "image %q", name)
 		}
 
-		// A reference to a piece of content has been removed,
-		// mark content store as dirty for triggering garbage
-		// collection
-		s.db.dirtyL.Lock()
-		s.db.dirtyCS = true
-		s.db.dirtyL.Unlock()
+		atomic.AddUint32(&s.db.dirty, 1)
```

Comment:
missing the dirtyCS setting here? even though the scanRoot will mark the manifest as removable.

Reply:
`dirtyCS` should only be set when content is actually removed, whether through GC or explicit deletion. Usually this should be done through the GC. This was called previously to force the GC to run, but the content store GC may have been unnecessary if the manifest has another reference.

Reply:
OK. `dirty` field can force to run gc.